### ISSUE TITLE
Using a key binding to move outside a bracket gives a wrong bracket highlighting.

### ISSUE THREAD

**Issue by jfcherng** [facelessuser/BracketHighlighter#260]
## Issue

Using a key binding to move outside a bracket gives a wrong bracket highlighting.

Initial status:
![init](https://cloud.githubusercontent.com/assets/6594915/8904149/9e5af16a-348f-11e5-9c25-86e68014af75.png)

Press `Ctrl+.` (a key binding in the `Environments` section) to move to the right bracket:
![right-key](https://cloud.githubusercontent.com/assets/6594915/8904150/9e5cb356-348f-11e5-8b78-cd79566fa5d3.png)

Expected result (Press right arrow to move the cursor one char right):
![right-arrow](https://cloud.githubusercontent.com/assets/6594915/8904151/9e5cecea-348f-11e5-90d8-36cff01b491d.png)

## Environments

`bracket_outside_adjacent` is set to `false` in plugin settings.
```javascript
    // Outside adjacent bracket matching
    "bracket_outside_adjacent": false,
```
And a key binding to move the cursor to the right bracket.
```javascript
{
    "keys": ["ctrl+."],
    "command": "bh_key",
    "args": {
        "no_outside_adj": true,
        "lines" : true,
        "plugin": {
            "type": ["__all__"],
            "command": "bh_modules.bracketselect",
            "args": {
                "select": "right"
            }
        }
    }
},
```

**Comment by facelessuser** [comment]
I understand the desire for the parent to be match in this scenario, but BH's current algorithm will not allow this without chaining multiple commands.  That is because the inner bracket was used as the match when calculating the jump outside the brackets.  The matching bracket is **never** aware of the parent bracket as BH only searches one bracket at a time.

So for clarification, the algorithm goes like this (highlight in bold):

1 . Match bracket:  ( **[** |**]** )
2. Move cursor: ( **[** **]**| )

The only way to end with **(** [ ]| **)** is if BH did the "jump outside" call **and** then called a command to do a secondary match after the move.

Commands are set up to do one thing and that is what they do.  The alternative would be to have it highlight nothing, but it can't highlight the parent without calling an additional match command  If you **must** have it highlight the command, I suggest writing a command wrapper that chains the jump with a secondary highlight.

**Comment by facelessuser** [comment]
**Maybe** I could expose something in the bh_plugin API that would trigger an auto match...let me think about this more.

**Comment by facelessuser** [comment]
I have a fix for this which will be released in 2.9.0 whenever I get around to it.  Probably some time this mid to late August.
